### PR TITLE
DAO light version

### PR DESCRIPTION
**Pull request by LouiseMedova**
- removed `admin`,  `membership_proposal`, votes delegation and some functions related to admin actions.
- added `deposit` function through which  everyone can become the DAO member

### CODE REVIEW

File: dao-light/tests/utils.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    let res = ft.send(
+        MEMBERS[0],
+        InitConfig {
+            name: String::from("MyToken"),
+            symbol: String::from("MTK"),
+        },
+    );
+
+    assert!(res.log().is_empty());
+    MEMBERS.iter().enumerate().for_each(|(_, member)| {
```
Comment (shamilsan) [inline dao-light/tests/utils.rs:24]: Why do we enumerate and don't use the index?
```suggestion
    MEMBERS.iter().for_each(|member| {
```

File: dao-light/tests/utils.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            details: "Funding proposal".to_string(),
+        },
+    )
+}
+
+pub fn vote(dao: &Program, member: u64, proposal_id: u128, vote: Vote) -> RunResult {
+    dao.send(
+        member,
+        DaoAction::SubmitVote {
+            proposal_id: proposal_id.clone(),
```
Comment (shamilsan) [inline dao-light/tests/utils.rs:65]: `u128` seems to be `Copy` type.
```suggestion
            proposal_id,
```

File: dao-light/tests/utils.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            vote: vote.clone(),
+        },
+    )
+}
+
+pub fn process(dao: &Program, member: u64, proposal_id: u128) -> RunResult {
+    dao.send(
+        member,
+        DaoAction::ProcessProposal {
+            proposal_id: proposal_id.clone(),
```
Comment (shamilsan) [inline dao-light/tests/utils.rs:75]: The same.
```suggestion
            proposal_id,
```

**Review by shamilsan — APPROVED** [review]
Great excepting minor issues. But need to commit the updated root `Cargo.lock` too!

**Review by gshep — APPROVED** [review]

File: dao-light/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+        period_duration: config.period_duration,
+        ..Dao::default()
+    };
+    DAO = Some(dao);
+}
+
+#[gstd::async_main]
+async unsafe fn main() {
+    let action: DaoAction = msg::load().expect("Could not load Action");
+    let dao: &mut Dao = unsafe { DAO.get_or_insert(Dao::default()) };
```
[the remark this answers is not in the document]
Comment (LouiseMedova) [reply dao-light/src/lib.rs:381]: It doesn't work without unsafe block

**Pull request merged**